frontend/src/app/shared/form/manifest-builder/manifest-attributes-array/manifest-attributes-array.component.ts

```diff
@@ -0,0 +1,167 @@
+    BoxTitleDirective,
+    PaginationComponent,
+    ValErrorDelayDirective,
+    ValidationErrorsComponent,
+    NgbTooltip
+  ]
+})
+export class ManifestAttributesArrayComponent {
+  private modalService = inject(ModalService);
+  private translateService = inject(TranslateService);
+
+  scanModes = input.required<Array<ScanModeDTO>>();
+  certificates = input.required<Array<CertificateDTO>>();
+  parentGroup = input.required<FormGroup>();
+  control = input.required<FormControl<Array<any>>>();
+  arrayAttribute = input.required<OIBusArrayAttribute>();
+
+  private readonly controlValue = toSignal(toObservable(this.control).pipe(switchMap(c => c.valueChanges.pipe(startWith(c.value)))));
+  readonly columns = computed(() => this.buildColumn(this.arrayAttribute().rootAttribute, []));
+  readonly paginatedValues = computed(() => {
+    if (this.arrayAttribute().paginate) {
+      return new ArrayPage(this.controlValue()!, this.arrayAttribute().numberOfElementPerPage);
+    }
+    return new ArrayPage([], 1);
+  });
+
+  async addItem(event: Event) {
+    event.preventDefault();
+    const modal = this.modalService.open(ManifestAttributeEditorModalComponent, { size: 'lg' });
+    modal.componentInstance.prepareForCreation(this.scanModes(), this.certificates());
+
+    modal.result.subscribe(arrayElement => {
+      this.control().setValue([...this.control().value, arrayElement]);
+      this.paginatedValues().gotoPage(0);
+    });
+  }
+
+  async copyItem(element: any) {
+    const modal = this.modalService.open(ManifestAttributeEditorModalComponent, { size: 'lg' });
+    modal.componentInstance.prepareForCreation(this.scanModes(), this.certificates());
+
+    // Pre-populate with the copied attribute's values
+    const copiedAttribute = { ...element, key: element.key + '_copy' };
+    modal.componentInstance.prepareForEdition(this.scanModes(), this.certificates(), copiedAttribute);
+
+    modal.result.subscribe(arrayElement => {
+      this.control().setValue([...this.control().value, arrayElement]);
+      this.paginatedValues().gotoPage(0);
+    });
+  }
+
+  async editItem(element: any) {
+    const modal = this.modalService.open(ManifestAttributeEditorModalComponent, { size: 'lg' });
+    modal.componentInstance.prepareForEdition(this.scanModes(), this.certificates(), element);
+
+    modal.result.subscribe(arrayElement => {
+      const newArray = [...this.control().value];
+      const index = this.control().value.indexOf(element);
+      newArray[index] = { ...arrayElement, id: element.id };
+
+      this.control().setValue(newArray);
+      this.paginatedValues().gotoPage(0);
+    });
+  }
+
+  deleteItem(element: any) {
+    const newArray = [...this.control().value];
+    const index = this.control().value.indexOf(element);
+    newArray.splice(index, 1);
+    this.control().setValue(newArray);
+    this.paginatedValues().gotoPage(0);
+  }
+
+  buildColumn(attribute: OIBusObjectAttribute, prefix: Array<string>): Array<Column> {
```

Comment:
buildColumn, formatValue and getValueByPath could be in a form-utils file, and these methods will be used from oibus-array-form-control. It will avoid code duplication